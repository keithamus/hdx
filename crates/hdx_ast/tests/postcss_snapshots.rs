mod helpers;

#[test]
#[cfg(feature = "serde")]
fn postcss_media() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/apply.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_brackets() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/at-rule-brackets.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_decls() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-decls.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_empty() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-empty.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_no_params() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-no-params.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_no_semicolon() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-no-semicolon.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_params() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-params.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_atrule_rules() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/atrule-rules.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_between() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/between.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_colon_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/colon-selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_comments() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/comments.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_custom_properties() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/custom-properties.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_decls() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/decls.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_empty() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/empty.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_escape() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/escape.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_extends() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/extends.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_function() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/function.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_ie_prodid() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/ie-progid.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_important() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/important.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_inside() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/inside.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_no_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/no-selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_prop() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/prop.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_quotes() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/quotes.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_raw_decl() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/raw-decl.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_rule_at() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/rule-at.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_rule_no_semicolon() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/rule-no-semicolon.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_selector() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/selector.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_semicolons() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/semicolons.css");
}

#[test]
#[cfg(feature = "serde")]
fn postcss_tab() {
	assert_snap_ast!("../../tasks/coverage/postcss-parser-tests/cases/tab.css");
}
