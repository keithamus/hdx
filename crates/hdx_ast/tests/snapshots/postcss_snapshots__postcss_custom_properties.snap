---
source: crates/hdx_ast/tests/postcss_snapshots.rs
expression: result.output.unwrap()
---
{
  "type": "stylesheet",
  "rules": [
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "pseudo-class",
              "value": {
                "active": [
                  {
                    "kind": "Colon",
                    "offset": 0,
                    "len": 1
                  },
                  {
                    "kind": "Ident",
                    "offset": 1,
                    "len": 4
                  }
                ]
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 6,
        "len": 1
      },
      "declarations": [
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 10,
            "len": 8
          },
          "colon": {
            "kind": "Colon",
            "offset": 18,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 19,
                "len": 1
              },
              {
                "kind": "Number",
                "offset": 20,
                "len": 1
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 21,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 25,
            "len": 6
          },
          "colon": {
            "kind": "Colon",
            "offset": 31,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 32,
                "len": 1
              },
              {
                "kind": "Dimension",
                "offset": 33,
                "len": 5,
                "unit": "vw"
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 38,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 42,
            "len": 7
          },
          "colon": {
            "kind": "Colon",
            "offset": 49,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 50,
                "len": 1
              },
              {
                "kind": "Hash",
                "offset": 51,
                "len": 4
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 55,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 60,
            "len": 10
          },
          "colon": {
            "kind": "Colon",
            "offset": 70,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 71,
                "len": 1
              },
              {
                "type": "Function",
                "name": {
                  "kind": "Function",
                  "offset": 72,
                  "len": 5
                },
                "values": [
                  {
                    "kind": "Number",
                    "offset": 77,
                    "len": 1
                  },
                  {
                    "kind": "Whitespace",
                    "offset": 78,
                    "len": 1
                  },
                  {
                    "kind": "Delim",
                    "offset": 79,
                    "len": 1
                  },
                  {
                    "kind": "Whitespace",
                    "offset": 80,
                    "len": 1
                  },
                  {
                    "kind": "Number",
                    "offset": 81,
                    "len": 1
                  }
                ],
                "close_paren": {
                  "kind": "RightParen",
                  "offset": 82,
                  "len": 1
                }
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 83,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 87,
            "len": 10
          },
          "colon": {
            "kind": "Colon",
            "offset": 97,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 98,
                "len": 1
              },
              {
                "type": "Function",
                "name": {
                  "kind": "Function",
                  "offset": 99,
                  "len": 4
                },
                "values": [
                  {
                    "kind": "Ident",
                    "offset": 103,
                    "len": 6
                  }
                ],
                "close_paren": {
                  "kind": "RightParen",
                  "offset": 109,
                  "len": 1
                }
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 110,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 115,
            "len": 8
          },
          "colon": {
            "kind": "Colon",
            "offset": 123,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 124,
                "len": 1
              },
              {
                "kind": "String",
                "offset": 125,
                "len": 22
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 147,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 151,
            "len": 8
          },
          "colon": {
            "kind": "Colon",
            "offset": 159,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 160,
                "len": 1
              },
              {
                "kind": "String",
                "offset": 161,
                "len": 22
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 183,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 188,
            "len": 14
          },
          "colon": {
            "kind": "Colon",
            "offset": 202,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 203,
                "len": 1
              },
              {
                "type": "SimpleBlock",
                "start": 204,
                "open": {
                  "kind": "LeftSquare",
                  "len": 1
                },
                "values": {
                  "values": [
                    {
                      "kind": "Number",
                      "offset": 205,
                      "len": 1
                    },
                    {
                      "kind": "Comma",
                      "offset": 206,
                      "len": 1
                    },
                    {
                      "kind": "Whitespace",
                      "offset": 207,
                      "len": 1
                    },
                    {
                      "kind": "Number",
                      "offset": 208,
                      "len": 1
                    },
                    {
                      "kind": "Comma",
                      "offset": 209,
                      "len": 1
                    },
                    {
                      "kind": "Whitespace",
                      "offset": 210,
                      "len": 1
                    },
                    {
                      "kind": "Number",
                      "offset": 211,
                      "len": 1
                    }
                  ]
                },
                "close": {
                  "kind": "RightSquare",
                  "len": 1
                }
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 213,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 217,
            "len": 13
          },
          "colon": {
            "kind": "Colon",
            "offset": 230,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 231,
                "len": 1
              },
              {
                "type": "SimpleBlock",
                "start": 232,
                "open": {
                  "kind": "LeftParen",
                  "len": 1
                },
                "values": {
                  "values": [
                    {
                      "kind": "Number",
                      "offset": 233,
                      "len": 1
                    },
                    {
                      "kind": "Comma",
                      "offset": 234,
                      "len": 1
                    },
                    {
                      "kind": "Whitespace",
                      "offset": 235,
                      "len": 1
                    },
                    {
                      "kind": "Number",
                      "offset": 236,
                      "len": 1
                    },
                    {
                      "kind": "Comma",
                      "offset": 237,
                      "len": 1
                    },
                    {
                      "kind": "Whitespace",
                      "offset": 238,
                      "len": 1
                    },
                    {
                      "kind": "Number",
                      "offset": 239,
                      "len": 1
                    }
                  ]
                },
                "close": {
                  "kind": "RightParen",
                  "len": 1
                }
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 241,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 246,
            "len": 8
          },
          "colon": {
            "kind": "Colon",
            "offset": 254,
            "len": 1
          },
          "value": {
            "values": []
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 256,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 260,
            "len": 8
          },
          "colon": {
            "kind": "Colon",
            "offset": 268,
            "len": 1
          },
          "value": {
            "values": []
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 274,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 278,
            "len": 8
          },
          "colon": {
            "kind": "Colon",
            "offset": 286,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 287,
                "len": 1
              },
              {
                "kind": "Delim",
                "offset": 288,
                "len": 1
              },
              {
                "kind": "Ident",
                "offset": 289,
                "len": 9
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 298,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 302,
            "len": 8
          },
          "colon": {
            "kind": "Colon",
            "offset": 310,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 315,
                "len": 1
              },
              {
                "kind": "Delim",
                "offset": 316,
                "len": 1
              },
              {
                "kind": "Ident",
                "offset": 317,
                "len": 9
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 326,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 330,
            "len": 8
          },
          "colon": {
            "kind": "Colon",
            "offset": 338,
            "len": 1
          },
          "value": {
            "values": []
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 344,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 349,
            "len": 15
          },
          "colon": {
            "kind": "Colon",
            "offset": 364,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 365,
                "len": 1
              },
              {
                "type": "SimpleBlock",
                "start": 366,
                "open": {
                  "kind": "LeftCurly",
                  "len": 1
                },
                "values": {
                  "values": [
                    {
                      "kind": "Number",
                      "offset": 367,
                      "len": 1
                    },
                    {
                      "kind": "Comma",
                      "offset": 368,
                      "len": 1
                    },
                    {
                      "kind": "Whitespace",
                      "offset": 369,
                      "len": 1
                    },
                    {
                      "kind": "Number",
                      "offset": 370,
                      "len": 1
                    },
                    {
                      "kind": "Comma",
                      "offset": 371,
                      "len": 1
                    },
                    {
                      "kind": "Whitespace",
                      "offset": 372,
                      "len": 1
                    },
                    {
                      "kind": "Number",
                      "offset": 373,
                      "len": 1
                    }
                  ]
                },
                "close": {
                  "kind": "RightCurly",
                  "len": 1
                }
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 375,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 379,
            "len": 6
          },
          "colon": {
            "kind": "Colon",
            "offset": 385,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 386,
                "len": 1
              },
              {
                "type": "SimpleBlock",
                "start": 387,
                "open": {
                  "kind": "LeftSquare",
                  "len": 1
                },
                "values": {
                  "values": [
                    {
                      "kind": "Number",
                      "offset": 388,
                      "len": 1
                    },
                    {
                      "kind": "Comma",
                      "offset": 389,
                      "len": 1
                    },
                    {
                      "kind": "Whitespace",
                      "offset": 390,
                      "len": 1
                    },
                    {
                      "kind": "String",
                      "offset": 391,
                      "len": 3
                    },
                    {
                      "kind": "Comma",
                      "offset": 394,
                      "len": 1
                    },
                    {
                      "kind": "Whitespace",
                      "offset": 395,
                      "len": 1
                    },
                    {
                      "type": "SimpleBlock",
                      "start": 396,
                      "open": {
                        "kind": "LeftCurly",
                        "len": 1
                      },
                      "values": {
                        "values": [
                          {
                            "kind": "String",
                            "offset": 397,
                            "len": 7
                          },
                          {
                            "kind": "Colon",
                            "offset": 404,
                            "len": 1
                          },
                          {
                            "kind": "Whitespace",
                            "offset": 405,
                            "len": 1
                          },
                          {
                            "type": "SimpleBlock",
                            "start": 406,
                            "open": {
                              "kind": "LeftCurly",
                              "len": 1
                            },
                            "values": {
                              "values": [
                                {
                                  "kind": "String",
                                  "offset": 407,
                                  "len": 3
                                },
                                {
                                  "kind": "Colon",
                                  "offset": 410,
                                  "len": 1
                                },
                                {
                                  "kind": "Number",
                                  "offset": 411,
                                  "len": 1
                                }
                              ]
                            },
                            "close": {
                              "kind": "RightCurly",
                              "len": 1
                            }
                          }
                        ]
                      },
                      "close": {
                        "kind": "RightCurly",
                        "len": 1
                      }
                    },
                    {
                      "kind": "Comma",
                      "offset": 414,
                      "len": 1
                    },
                    {
                      "kind": "Whitespace",
                      "offset": 415,
                      "len": 1
                    },
                    {
                      "type": "SimpleBlock",
                      "start": 416,
                      "open": {
                        "kind": "LeftSquare",
                        "len": 1
                      },
                      "values": {
                        "values": [
                          {
                            "kind": "Number",
                            "offset": 417,
                            "len": 1
                          }
                        ]
                      },
                      "close": {
                        "kind": "RightSquare",
                        "len": 1
                      }
                    }
                  ]
                },
                "close": {
                  "kind": "RightSquare",
                  "len": 1
                }
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 420,
            "len": 1
          }
        },
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 424,
            "len": 12
          },
          "colon": {
            "kind": "Colon",
            "offset": 436,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Whitespace",
                "offset": 437,
                "len": 1
              },
              {
                "type": "Function",
                "name": {
                  "kind": "Function",
                  "offset": 438,
                  "len": 9
                },
                "values": [
                  {
                    "kind": "Ident",
                    "offset": 447,
                    "len": 4
                  }
                ],
                "close_paren": {
                  "kind": "RightParen",
                  "offset": 451,
                  "len": 1
                }
              },
              {
                "kind": "Whitespace",
                "offset": 452,
                "len": 1
              },
              {
                "type": "SimpleBlock",
                "start": 453,
                "open": {
                  "kind": "LeftCurly",
                  "len": 1
                },
                "values": {
                  "values": [
                    {
                      "kind": "Whitespace",
                      "offset": 454,
                      "len": 1
                    },
                    {
                      "kind": "Ident",
                      "offset": 455,
                      "len": 7
                    },
                    {
                      "kind": "Delim",
                      "offset": 462,
                      "len": 1
                    },
                    {
                      "type": "Function",
                      "name": {
                        "kind": "Function",
                        "offset": 463,
                        "len": 4
                      },
                      "values": [
                        {
                          "kind": "Ident",
                          "offset": 467,
                          "len": 4
                        }
                      ],
                      "close_paren": {
                        "kind": "RightParen",
                        "offset": 471,
                        "len": 1
                      }
                    }
                  ]
                },
                "close": {
                  "kind": "RightCurly",
                  "len": 1
                }
              }
            ]
          },
          "important": null,
          "semicolon": {
            "kind": "Semicolon",
            "offset": 474,
            "len": 1
          }
        }
      ],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 476,
        "len": 1
      }
    },
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "pseudo-class",
              "value": {
                "active": [
                  {
                    "kind": "Colon",
                    "offset": 479,
                    "len": 1
                  },
                  {
                    "kind": "Ident",
                    "offset": 480,
                    "len": 4
                  }
                ]
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 484,
        "len": 1
      },
      "declarations": [
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 485,
            "len": 3
          },
          "colon": {
            "kind": "Colon",
            "offset": 488,
            "len": 1
          },
          "value": {
            "values": [
              {
                "kind": "Number",
                "offset": 489,
                "len": 1
              }
            ]
          },
          "important": null,
          "semicolon": null
        }
      ],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 490,
        "len": 1
      }
    },
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "class",
              "value": {
                "type": "Class",
                "dot": {
                  "kind": "Delim",
                  "offset": 493,
                  "len": 1
                },
                "name": {
                  "kind": "Ident",
                  "offset": 494,
                  "len": 14
                }
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 509,
        "len": 1
      },
      "declarations": [
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 510,
            "len": 11
          },
          "colon": {
            "kind": "Colon",
            "offset": 521,
            "len": 1
          },
          "value": {
            "values": []
          },
          "important": null,
          "semicolon": null
        }
      ],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 527,
        "len": 1
      }
    },
    {
      "type": "stylerule",
      "selectors": [
        {
          "components": [
            {
              "type": "class",
              "value": {
                "type": "Class",
                "dot": {
                  "kind": "Delim",
                  "offset": 530,
                  "len": 1
                },
                "name": {
                  "kind": "Ident",
                  "offset": 531,
                  "len": 5
                }
              }
            }
          ],
          "comma": null
        }
      ],
      "type": "style-declaration",
      "open": {
        "kind": "LeftCurly",
        "offset": 536,
        "len": 1
      },
      "declarations": [
        {
          "type": "property",
          "name": {
            "kind": "Ident",
            "offset": 537,
            "len": 3
          },
          "colon": {
            "kind": "Colon",
            "offset": 540,
            "len": 1
          },
          "value": {
            "values": []
          },
          "important": null,
          "semicolon": null
        }
      ],
      "rules": [],
      "close": {
        "kind": "RightCurly",
        "offset": 542,
        "len": 1
      }
    }
  ]
}
